### DISCUSSION TITLE
mvt-layer can't display sublayer normally

### DISCUSSION THREAD

**Discussion by yanshjie** [visgl/deck.gl#7245] [Q&A] [answered]
Hello, I try to use MVTLayer's example to load the map, the coordinates of mvt data parsing are Cartesian coordinates: [0.292236328125, 0.811279296875], [0.29833984375, 0.81689453125]. 
When I want to use **rendersubLayers** to redraw the mvt data, I found that this type of coordinates can't support geojsonlayer to display data normally. I guess it may because of different coordinate support systems but I still want to know if there's anything to resolve this problem when using sublayer?
![lQLPJxapzx5juG3NASvNAjSwboKL44fJxtoDF0gNzsDSAA_564_299 png_620x10000q90](https://user-images.githubusercontent.com/52058277/188805778-2d9cdc79-37cf-44d1-8a4e-4f8b552f8da2.jpg)

this is the code
```
   new MVTLayer({
    id: "street",
    data: [
      '[https://tiles-a.basemaps.cartocdn.com/vectortiles/carto.streets/v1/{z}/{x}/{y}.mvt](https://tiles-a.basemaps.cartocdn.com/vectortiles/carto.streets/v1/%7Bz%7D/%7Bx%7D/%7By%7D.mvt)'
    ],
    binary: false,
    minZoom: 0,
    maxZoom: 23,
    getLineColor: [192, 192, 192],
    getFillColor: [140, 170, 180],
    getLineWidth: f => {
      switch (f.properties.class) {
        case 'street':
        return 6;
        case 'motorway':
        return 10;
        default:
        return 1;
      }
    },
    lineWidthMinPixels: 1  ,
    renderSubLayers: (props) => {
       let allData = [];
       props.data.forEach((feature) => {
         allData.push(feature); 
       });
       const myLayer = new GeoJsonLayer({
         id: props.id + "geojson",
         data: props.data,
         filled: true,
         getFillColor: [161, 203, 234],
       });
       return myLayer
     }
  })
```

**Comment by yanshjie** [comment]
It seems we use rendersublayers incorrectly. The upper code in the picture works but the rest can‘t work
![图片](https://user-images.githubusercontent.com/98957105/188837681-215e97d7-8d9c-474b-bb00-86b698796755.png)

**Comment by ibgreen** [comment, marked as answer]
You are not passing all the `props` in the lower case.  Maybe inspect it in the debugger and check what you are missing?

Or add `{...props, id: ...}` to your params make sure you copy them all.
